File: willfix-bytex/src/main/java/wtf/s1/willfix/bytex/WillFixByteXExtension.java
package wtf.s1.willfix.bytex;

import com.ss.android.ugc.bytex.common.BaseExtension;

import java.util.List;

public class WillFixByteXExtension extends BaseExtension {
    @Override
    public String getName() {
        return "willFixByteX";
    }

    public List<String> methodList;

    public String separator = "#";

    public String exceptionHandler;

    public List<String> getMethodList() {
        return methodList;
    }

    public void methodList(List<String> methodList) {
        this.methodList = methodList;
    }

    public String getSeparator() {
        return separator;
    }

    public void separator(String separator) {
        this.separator = separator;
    }

    public String getExceptionHandler() {
        return exceptionHandler;
    }

    public void exceptionHandler(String catchHandler) {
        this.exceptionHandler = catchHandler;
    }
}

File: willfix-bytex/src/main/java/wtf/s1/willfix/bytex/WillFixByteXExtension.kt
package wtf.s1.willfix.bytex

import com.ss.android.ugc.bytex.common.BaseExtension

open class WillFixByteXExtension : BaseExtension() {
    override fun getName(): String {
        return "willFixByteX"
    }

    var methodList: List<String>? = null
    var separator = "#"
    var exceptionHandler: String? = null

    fun methodList(methodList: List<String>?) {
        this.methodList = methodList
    }

    fun separator(separator: String) {
        this.separator = separator
    }

    fun exceptionHandler(catchHandler: String?) {
        exceptionHandler = catchHandler
    }
}